source/Octopus.Tentacle/Kubernetes/KubernetesScriptPodCreator.cs

```diff
@@ -318,58 +320,58 @@ V1ResourceRequirements GetScriptPodResourceRequirements(InMemoryTentacleScriptLo
         }
 
         V1Affinity ParseScriptPodAffinity(InMemoryTentacleScriptLog tentacleScriptLog)
+            => ParseScriptPodJson(
+                tentacleScriptLog,
+                KubernetesConfig.PodAffinityJson,
+                KubernetesConfig.PodAffinityJsonVariableName,
+                "pod affinity",
+                //we default to running on linux/arm64 and linux/amd64 nodes
+                new V1Affinity(new V1NodeAffinity(requiredDuringSchedulingIgnoredDuringExecution: new V1NodeSelector(new List<V1NodeSelectorTerm>
+                {
+                    new(matchExpressions: new List<V1NodeSelectorRequirement>
+                    {
+                        new("kubernetes.io/os", "In", new List<string> { "linux" }),
+                        new("kubernetes.io/arch", "In", new List<string> { "arm64", "amd64" })
+                    })
+                }))))!;
+
+        List<V1Toleration>? ParseScriptPodTolerations(InMemoryTentacleScriptLog tentacleScriptLog)
+            => ParseScriptPodJson<List<V1Toleration>>(
+                tentacleScriptLog,
+                KubernetesConfig.PodTolerationsJson,
+                KubernetesConfig.PodTolerationsJsonVariableName,
+                "pod tolerations");
+
+        V1PodSecurityContext? ParseScriptPodSecurityContext(InMemoryTentacleScriptLog tentacleScriptLog)
+            => ParseScriptPodJson<V1PodSecurityContext>(
+                tentacleScriptLog,
+                KubernetesConfig.PodSecurityContextJson,
+                KubernetesConfig.PodSecurityContextJsonVariableName,
+                "pod security context");
+        
+        [return: NotNullIfNotNull("defaultValue")]
```

Comment:
I didn't know this existed - that's neat. What happens if the Pod security context is null rather than an empty object in this case though? Will the pod creation fail?

Reply:
I'll double check, but I'm pretty confident that property is nullable right now and because we weren't setting it, was just getting `null`.